include/SampleFrame.h

```diff
@@ -226,6 +231,18 @@ inline void copyFromSampleFrames(float* target, const SampleFrame* source, size_
 	}
 }
 
+// Enable SampleFrame to use the AudioDataType metafunction
+namespace detail {
+template<> struct AudioDataType<AudioDataKind::SampleFrame> { using type = SampleFrame; };
+} // namespace detail
```

Comment:
I believe one of these specializations were in `AudioData.h`, and another one is here. Why not just put them all in `AudioData.h`?

Reply:
I wasn't sure if it would work due to `SampleFrame` not being defined in this header, but maybe it would work with a forward declaration

Reply:
Ah okay, didn't think about that. Yeah try a forward declaration since it feels neater to put them all together in `AudioData.h`.

Reply:
It worked with the forward declaration.

Done in https://github.com/LMMS/lmms/pull/7459/commits/f3a8b3f3f33d1b06dca907f24ae98e2ae2c0d185